src/python/WMCore/MicroService/Unified/MSMonitor.py

```diff
@@ -0,0 +1,190 @@
+            self.logger.debug('+++ monit found %s requests in %s state',
+                              len(requests), reqStatus)
+
+            requestStatuses = {}  # keep track of request statuses
+            for reqName in requests:
+                req = {'name': reqName, 'reqStatus': reqStatus}
+                self.logger.debug("+++ request %s", req)
+                # obtain status records from couchdb for given request
+                transferRecords = self.getTransferRecords(reqName)
+                completion = self.completion(transferRecords)
+                # if all transfers are completed
+                # move the request status staging -> staged
+                if completion == self.msConfig.get('completedThreshold', 100):
+                    self.logger.debug(
+                        "+++ request %s all transfers are completed", req)
+                    self.change(req, 'staged', '+++ monit')
+                # if pileup transfers are completed AND
+                # some input blocks are completed,
+                # move the request status staging -> staged
+                elif self.pileupTransfersCompleted(transferRecords):
+                    self.logger.debug(
+                        "+++ request %s pileup transfers are completed", req)
+                    self.change(req, 'staged', '+++ monit')
+                # transfers not completed
+                # just update the database with their completion
+                else:
+                    self.logger.debug(
+                        "+++ request %s transfers are %s completed",
+                        req, completion)
+                    # for us workflow name is the same as request name
+                    wname = reqName
+                    requestStatuses[wname] = transferRecords
+            self.updateTransferInfo(requestStatuses)
+        except Exception as err:  # general error
+            self.logger.exception('+++ monit error: %s', str(err))
+
+    def getTransferRecords(self, reqName):
+        """
+        Get transfer records for given request name from CouchDB.
+        Transfer records on backend has the following form
+        https://gist.github.com/amaltaro/72599f995b37a6e33566f3c749143154
+        So far logic is based on option D of the records
+
+        .. doctest::
+
+            {"workflowName": "bla",
+             "timestamp": 123,
+             "transfers": [rec1, rec2, ... ]
+            }
+            where each record has the following format:
+            {"dataset":"/a/b/c", "dataType": "primary", "transferIDs": [1,2], "completion":0}
+
+        :param reqName: name of the request
+        :return: list of of status records
+        """
+        transferRecords = []
+
+        # in our case workflow name is the same as request name
+        wname = reqName
+        # get existing transfer IDs record
+        doc = self.getTransferInfo(wname)
+        # loop over all records and obtain new transfer IDs
```

Comment:
```
if not doc:
   self.logger.error("Failed to find a transfer document for request: %s.", wname)
   return transferRecords
```

or something like that. Then we need to skip this workflow on the for loop calling this function.

Reply:
done